ego4d/internal/validation/validate.py

```diff
@@ -1478,6 +1478,96 @@ def validate_egoexo_files(
+
+    def get_post_survey_paths(captures: List[CaptureMetadataEgoExo]) -> List[ReferencedFile]:
+        dict_post_surveys = {}
+        for capture in captures:
+            if not capture.post_surveys_relative_path.startswith("s3://"):
+                survey_path = os.path.join(capture.university_video_folder_path, capture.post_surveys_relative_path) # make absolute path if not allready
+            else:
+                survey_path = capture.post_surveys_relative_path
+            dict_post_surveys[capture.university_capture_id] = survey_path
+        return dict_post_surveys
+
+    def check_file_post_survey(file_path_dict):
+        for key, value in file_path_dict.items():
+            if not os.path.exists(value):
+                ret.append(
+                    Error(
+                        ErrorLevel.ERROR,
+                        value,
+                        "path_does_not_exist",
+                        "file path {value} does not exist",
+                    )
+                )
+        return ret
+
+    post_survey_files = get_post_survey_paths(manifest.captures.values())
+    ret.extend(check_file_post_survey(post_survey_files))
+    def open_csv(capture_id_file_path):
+        return capture_id_file_path[0], pd.read_csv(pathmgr.open(capture_id_file_path[1]), dtype=object)
+
+    dict_pandas_df = {}
+    with ThreadPoolExecutor(num_workers) as pool:
+        vals_to_map = list(post_survey_files.items())
+        map_fn = open_csv
+        print("Reading Post Survey Data...")
+        for capture_id, df in tqdm(pool.map(map_fn, vals_to_map), total=len(vals_to_map)):
+            dict_pandas_df[capture_id] = df
+
+# Matching logic below is based of Miguel's notebook: N4348396
+    all_column_ks = set()
+
+    for takes in manifest.takes.values():
+        for take in takes:
+            post_survey_data = dict_pandas_df[take.university_capture_id]
+
+            if post_survey_data is not None:
+                all_column_ks.update(set(post_survey_data.columns))
+                tid_key = [
+                    k for k in post_survey_data.columns
+                    if ("Take ID" in k) or ("Take_ID" in k) or ("take_id" in k)
+                ]
+                if len(tid_key) != 1:
+                    post_survey_data = None
+                    ret.append(
+                    Error(
+                        ErrorLevel.ERROR,
+                        participant_id,
+                        "participant_scenario_no_post_survey",
+                        "participant {p.participant_id} + take {take.university_capture_id} does not have a corresponding post_survey.",
+                    )
+                )
+                else:
+                    tid_key = tid_key[0]
+                    post_survey_data = post_survey_data[post_survey_data[tid_key] == take.take_id]
+                    post_survey_data = post_survey_data.iloc[0] if len(post_survey_data) > 0 else None
```

Comment:
Note: the reason to check `if post_survey_data is None` outside of the if/else block is due to this branch and line. After this line `post_survey_data` may be `None` since it may not match on the `take.take_id`